pkg/storage/unified/apistore/prepare.go

```diff
@@ -143,12 +143,10 @@ func (s *Storage) prepareObjectForUpdate(ctx context.Context, updateObject runti
 
 func (s *Storage) handleLargeResources(ctx context.Context, obj utils.GrafanaMetaAccessor, buf bytes.Buffer) ([]byte, error) {
 	support := s.opts.LargeObjectSupport
-	if support != nil {
```

Comment:
Seems off to me that we were deconstructing objects without considering whether their size exceeds the threshold. All we were enforcing is for the size not to exceed `support.MaxSize()`.